src/Microsoft.Windows.CsWin32/templates/HRESULT.cs

```diff
@@ -61,7 +61,9 @@ internal HRESULT ThrowOnFailure(IntPtr errorInfo = default)
         return this;
     }
 
-    public override string ToString() => this.Value.ToString();
+	internal static HRESULT HRESULT_FROM_WIN32(WIN32_ERROR error) => new(((int)error & 0x0000FFFF) | unchecked((int)0x80070000));
 
-    internal string ToString(string format, IFormatProvider formatProvider) => ((uint)this.Value).ToString(format, formatProvider);
+	public override string ToString() => this.Value.ToString("X");
```

Comment:
We want it in this format: `0x00005678`. That is, a `0x` prefix and padded so it ends with 8 hex characters.